cpp/CMakeLists.txt

```diff
@@ -15,9 +15,14 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.25)
 message(STATUS "Building using CMake version: ${CMAKE_VERSION}")
 
+# https://cmake.org/cmake/help/latest/policy/CMP0126.html
+#
+# set(CACHE) does not remove a normal variable of the same name.
```

Comment:
Do we plan to fix this issue? Which variable(s) are concerned?

Reply:
Yes, I added a comment on the PR that this can be tackled as a separate issue. There is an issue with `ZLIB_LIBRARY` on Windows when using bundled ORC. If we decide it's ok to merge and fix later, I'll open an issue to track it. I was mainly waiting to find if @kou knew what could be the solution as I just found out today :)

Reply:
Can we remove a normal `ZLIB_LIBRARY` by `unset` https://cmake.org/cmake/help/latest/command/unset.html explicitly instead?

```diff
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index 4ef9fa8f4c..8001f07d8e 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -4664,6 +4664,7 @@ function(build_orc)
     set(ZLIB_HOME
         ${ZLIB_ROOT}
         CACHE STRING "" FORCE)
+    set(unset ZLIB_LIBRARY)
     set(ZLIB_LIBRARY
         ZLIB::ZLIB
         CACHE STRING "" FORCE)
```

Reply:
This does not seem to fix the problem, see latest failure with this commit:
https://github.com/apache/arrow/pull/44989/commits/a4b9e1b9bbc7f2146d205e04dc1fa59da3536073

Reply:
Ah, sorry. It should be `unset(ZLIB_LIBRARY)`.